ctapipe/image/cleaning.py

```diff
@@ -241,6 +243,56 @@ def apply_time_delta_cleaning(
     return pixels_to_keep
 
 
+def apply_time_average_cleaning(
+    geom, mask, image, arrival_times, picture_thresh, time_limit
+):
+    """
+    Extract all pixels that arrived within a given timeframe 
+    with respect to the time average of the pixels on the main island.
```

Comment:
This function does more than the docstring says. It somehow adapts the time limit dependent on the image and on picture threshold.

Reply:
Thanks, I will change the docstring accordingly.